tests/unit/test_main.py

```diff
@@ -99,46 +103,126 @@ def test_log_or_exit_on_exceptions_debug(plugin_exc_mock, mocker, monkeypatch):
 
     errors = [('bad.plugin', plugin_exc_mock)]
 
-    main._log_or_exit_on_exceptions(errors, debug=True)
+    main._log_or_exit_on_exceptions('base msg', errors, debug=True)
 
     logging_mock.warn.assert_called_once()
     logging_mock.error.assert_not_called()
 
 
-run_args = 'has_active_plugins,exp_log_count'
-run_params = [
-    (True, 2),
-    (False, 1),
-]
+@pytest.fixture
+def mock_log_or_exit_on_exc(mocker, monkeypatch):
+    mock = mocker.Mock()
+    monkeypatch.setattr('gordon.main._log_or_exit_on_exceptions', mock)
+    return mock
+
+
+@pytest.mark.parametrize('plugins,exp_log_calls', (
+    (('event_consumer', 'enricher', 'publisher', 'runnable'), 0),
+    (('event_consumer', 'enricher', 'publisher'), 0),
+    (('event_consumer', 'enricher'), 1),
```

Comment:
We probably don't need this case if the enricher becomes optional.

Reply:
It's fine to have it - it's the same expected result, and no harm in an extra test.